### PR TITLE
[-] Performance - Improve the speed of listing the records by executing their count into another request

### PR DESCRIPTION
**Pull request by root-io**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by root-io** [comment]
[(13) Users want to dissociate count (meta) from data when listing all records](https://trello.com/c/Nms1rxL2/2666-13-users-want-to-dissociate-count-meta-from-data-when-listing-all-records)

File: src/routes/resources.js [hunk 1]
```diff
@@ -42,6 +36,15 @@ module.exports = function (app, model, Implementation, integrator, opts) {
       .catch(next);
   };
 
+  this.count = function (request, response, next) {
+    const params = request.query;
+
+    return new Implementation.ResourcesGetter(model, opts, params)
+      .perform('count')
```
Comment (arnaudbesnier) [inline src/routes/resources.js]: `.perform('count')` is not really beautiful. Why not just `.count()`?

File: src/routes/associations.js [hunk 2]
```diff
@@ -20,13 +20,19 @@ module.exports = function (app, model, Implementation, integrator, opts) {
   }
 
   function getAssociation(request) {
+    const pathSplit = request.route.path.split('/');
+    let associationName = pathSplit[pathSplit.length - 1];
+
+    if (['.csv', 'count'].includes(associationName)) {
+      associationName = pathSplit[pathSplit.length - 2];
```
Comment (arnaudbesnier) [inline src/routes/associations.js]: Are you sure you didn't break the CSV export here?

Comment (root-io) [reply src/routes/associations.js]: Thanks, fixed ! 🙈

File: src/routes/associations.js [hunk 3]
```diff
@@ -62,6 +65,22 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
   }
 
+  function count(request, response, next) {
+    var association = getAssociation(request);
+    var params = _.extend(request.query, request.params, association);
+    var models = Implementation.getModels();
+    var associationField = getAssociationField(params.associationName);
+    var associationModel = _.find(models, function (model) {
+      return Implementation.getModelName(model) === associationField;
+    });
```
Comment (arnaudbesnier) [inline src/routes/associations.js]: Duplicate code.

File: src/routes/associations.js [hunk 4]
```diff
@@ -62,6 +63,21 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
   }
 
+  function count(request, response, next) {
+    const association = getAssociation(request);
+    const params = _.extend(request.query, request.params, association);
+    const models = Implementation.getModels();
+    const associationField = getAssociationField(params.associationName);
+    const associationModel = _.find(models, function (model) {
+      return Implementation.getModelName(model) === associationField;
+    });
```
Comment (arnaudbesnier) [inline src/routes/associations.js]: Duplicate code.

File: src/serializers/resource.js [hunk 5]
```diff
@@ -7,8 +7,7 @@ const SmartFieldsValuesInjector = require('../services/smart-fields-values-injec
 const Schemas = require('../generators/schemas');
 const logger = require('../services/logger');
 
-function ResourceSerializer(Implementation, model, records, integrator, opts, meta,
-  fieldsSearched, searchValue, fieldsPerModel) {
+function ResourceSerializer(Implementation, model, records, integrator, opts, fieldsSearched,searchValue, fieldsPerModel) {
```
Comment (arnaudbesnier) [inline src/serializers/resource.js]: The ResourceSerializer is a "public" function exposed for the Smart Collection implementations for example. If people cannot add meta anymore, they won't have the count anymore.

What would happen to the people using the current signature in production if they upgrade to this version?

File: src/routes/associations.js [hunk 6]
```diff
@@ -20,32 +21,40 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
       .then(function (results) {
         var records = results[0];
-        var count = results[1];
-        var fieldsSearched = results[2];
+        var fieldsSearched = results[1];
 
-        var meta = {
-          count: count
-        };
+        const meta = null;
```
Comment (arnaudbesnier) [inline src/routes/associations.js]: why don't you just send null in the function call?

File: src/routes/resources.js [hunk 7]
```diff
@@ -17,12 +17,9 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
       .then(function (results) {
         const records = results[0];
-        const count = results[1];
-        const fieldsSearched = results[2];
+        const fieldsSearched = results[1];
 
-        const meta = {
-          count: count
-        };
+        const meta = null;
```
Comment (arnaudbesnier) [inline src/routes/resources.js]: Same here. Quite weird to read.

File: src/routes/associations.js (hunk 6 above)
Comment (root-io) [reply src/routes/associations.js]: done 💪

**Pull request merged**